eng/pipelines/templates/steps/publish-cli-winget.yml

```diff
@@ -0,0 +1,53 @@
+parameters:
+  - name: AllowPrerelease
+    type: boolean
+    default: true
+    displayName: Allow prerelease versions to be published. Should be "false" after GA.
+  - name: CLiVersion
+    type: string
+    default: $(CLI_VERSION)
+    displayName: CLI version
+  - name: GitHubToken
+    type: string
+    default: $(azuresdk-github-pat)
+    displayName: GitHub token
```

Comment:
Any reason to make this a parameter rather than just acquire it as you normally would a variable?

Reply:
This is a pattern I unthinkingly apply in a few cases. 

* It makes the yml file "more general" in that we can change the PAT from a "higher level" yaml file. For example, when releasing to storage accounts I can send the bits to a test storage account and leave the release storage account untouched. In the case of GitHub PATs, we probably will never change this value. 
* It can also keep the code DRY in some cases (though not in this case).